### PR TITLE
Fix kubelet log format flag error

### PR DESCRIPTION
**Pull request by yuzhiquan**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
- fix kubelet logging format msg error,the '\n' not affective because the ``.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig instrumentation
/triage accepted
/cc @serathius

### CODE REVIEW

**Comment by yuzhiquan** [comment]
Ping @serathius @yujuhong

File: staging/src/k8s.io/component-base/logs/registry.go [hunk 1]
```diff
@@ -103,4 +107,5 @@ func init() {
 	// Text format is default klog format
 	logRegistry.Register(defaultLogFormat, nil)
 	logRegistry.Register(jsonLogFormat, json.JSONLogger)
+	Formats = fmt.Sprintf(`"%s"`, strings.Join(logRegistry.List(), `", "`))
```
Comment (serathius) [inline staging/src/k8s.io/component-base/logs/registry.go:110]: logRegistry is meant for allowing runtime registering of formats. Creating list of formats in init is to early. We should not generate list of formats until formats.frozen is called.

Comment (yuzhiquan) [reply staging/src/k8s.io/component-base/logs/registry.go]: Rollback to hardcode for log format.

File: staging/src/k8s.io/component-base/logs/options.go [hunk 2]
```diff
@@ -84,9 +87,8 @@ func flagIsSet(name string) bool {
 
 // AddFlags add logging-format flag
 func (o *Options) AddFlags(fs *pflag.FlagSet) {
-	unsupportedFlags := fmt.Sprintf("--%s", strings.Join(unsupportedLoggingFlags(), ", --"))
 	formats := fmt.Sprintf(`"%s"`, strings.Join(logRegistry.List(), `", "`))
-	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf("Sets the log format. Permitted formats: %s.\nNon-default formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", formats, unsupportedFlags))
+	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf("Sets the log format. Permitted Formats: %s.\nNon-default Formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", formats, UnsupportedFlags))
```
Comment (serathius) [inline staging/src/k8s.io/component-base/logs/options.go:91]:
```suggestion
	fs.StringVar(&o.LogFormat, logFormatFlagName, defaultLogFormat, fmt.Sprintf("Sets the log format. Permitted formats: %s.\nNon-default formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", formats, UnsupportedFlags))
```
I don't think we need to capitalize word "formats"

File: cmd/kubelet/app/options/options.go [hunk 3]
```diff
@@ -544,7 +545,7 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
...
 	fs.StringVar(&c.KubeReservedCgroup, "kube-reserved-cgroup", c.KubeReservedCgroup, "Absolute name of the top level cgroup that is used to manage kubernetes components for which compute resources were reserved via '--kube-reserved' flag. Ex. '/kube-reserved'. [default='']")
-	fs.StringVar(&c.Logging.Format, "logging-format", c.Logging.Format, `Sets the log format. Permitted formats: "text", "json".\nNon-default formats don't honor these flags: -add_dir_header, --alsologtostderr, --log_backtrace_at, --log_dir, --log_file, --log_file_max_size, --logtostderr, --skip_headers, --skip_log_headers, --stderrthreshold, --log-flush-frequency.\nNon-default choices are currently alpha and subject to change without warning.`)
+	fs.StringVar(&c.Logging.Format, "logging-format", c.Logging.Format, fmt.Sprintf("Sets the log format. Permitted Formats: \"text\", \"json\".\nNon-default Formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", logs.UnsupportedFlags))
```
Comment (serathius) [inline cmd/kubelet/app/options/options.go:548]:
```suggestion
	fs.StringVar(&c.Logging.Format, "logging-format", c.Logging.Format, fmt.Sprintf("Sets the log format. Permitted formats: \"text\", \"json\".\nNon-default formats don't honor these flags: %s.\nNon-default choices are currently alpha and subject to change without warning.", logs.UnsupportedFlags))
```
No need to capitalize

**Comment by yuzhiquan** [comment]
/assign @yujuhong 
please review kubelet part

**Comment by yuzhiquan** [comment]
/assign @Random-Liu @dchen1107 
Would you like review kubelet part?

**Review by ffromani — COMMENTED** [review]
/lgtm
seems like a straightforward change from kubelet perspective

**Comment by yuzhiquan** [comment]
Rebased, but need-rebase label still there.

**Comment by SergeyKanzhelev** [comment]
> Rebased, but need-rebase label still there.

it happens sometimes... Let me try:

/remove-label needs-rebase

**Comment by SergeyKanzhelev** [comment]
trying to get rid or need-rebase by closing and reopening. 

/close

**Pull request merged**